cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -17,12 +17,37 @@ limitations under the License.
 package kubeadm
 
 import (
+	"crypto/x509"
+
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// PublicKeyAlgorithm is a type of encryption keys used for certificates generated by kubeadm
+type PublicKeyAlgorithm int
+
+const (
+	// RSA represents RSA algorithm
+	RSA PublicKeyAlgorithm = iota
+	// ECDSA represents ECDSA algorithm
+	ECDSA
```

Comment:
can we get away with using elements from the go enumeration directly?
https://golang.org/src/crypto/x509/x509.go

```
type PublicKeyAlgorithm int

const (
	UnknownPublicKeyAlgorithm PublicKeyAlgorithm = iota
	RSA
	DSA
	ECDSA
	Ed25519
)
```

e.g. `x509.ECDSA`